### PR TITLE
Script tasks

### PR DESCRIPTION
**Pull request by ericsciple**

### CODE REVIEW

File: Tasks/powerShell/powershell.ps1 [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        }
+
+        $agent = Get-VstsTaskVariable -Name 'agent.version'
+        if (!$agent -or (([version]$agent) -lt $Minimum)) {
+            Write-Error "Agent version $Minimum or higher is required."
+        }
+    }
+
+    # Get inputs.
+    $input_apartment = Get-VstsInput -Name 'apartment' -Default 'Sta'
```
Comment (TingluoHuang) [inline Tasks/powerShell/powershell.ps1]: maybe just hard code 'STA'

File: Tasks/powerShell/powershell.ps1 [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    $input_errorActionPreference = Get-VstsInput -Name 'errorActionPreference' -Default 'Stop'
+    switch ($input_errorActionPreference.ToUpperInvariant()) {
+        'STOP' { }
+        'CONTINUE' { }
+        'SILENTLYCONTINUE' { }
+        default {
+            Write-Error (Get-VstsLocString -Key 'PS_InvalidErrorActionPreference' -ArgumentList $input_errorActionPreference)
+        }
+    }
+    $input_executionPolicy = Get-VstsInput -Name 'executionPolicy' -Default 'Unrestricted'
```
Comment (TingluoHuang) [inline Tasks/powerShell/powershell.ps1]: hard code for now?

File: Tasks/powerShell/powershell.ps1 [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        'FileName' = $powershellPath
+        'Arguments' = $arguments
+        'WorkingDirectory' = $workingDirectory
+        'RequireExitCodeZero' = !$ignoreExitCode
+    }
+    if (!$input_failOnStderr) {
+        Invoke-VstsTool @splat
+    } else {
+        $global:ErrorActionPreference = 'Continue'
+        Invoke-VstsTool @splat 2>&1 |
```
Comment (TingluoHuang) [inline Tasks/powerShell/powershell.ps1]: we need ##vso[task.setresult failure];

File: Tasks/powerShell/powershell.ps1 [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+    # Run the script.
+    $powershellPath = (Get-Command -Name powershell.exe -CommandType Application).Path
+    Assert-VstsPath -LiteralPath $powershellPath -PathType 'Leaf'
+    $arguments = "-NoLogo -$input_apartment -NoProfile -NonInteractive -ExecutionPolicy $input_executionPolicy -File `"$filePath`""
+    $splat = @{
+        'FileName' = $powershellPath
+        'Arguments' = $arguments
+        'WorkingDirectory' = $workingDirectory
+        'RequireExitCodeZero' = !$ignoreExitCode
```
Comment (TingluoHuang) [inline Tasks/powerShell/powershell.ps1]: input_ignoreExitCode

File: Tasks/powerShell/powershell.ps1 [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        ([System.Text.Encoding]::UTF8))
+
+    # Run the script.
+    $powershellPath = (Get-Command -Name powershell.exe -CommandType Application).Path
+    Assert-VstsPath -LiteralPath $powershellPath -PathType 'Leaf'
+    $arguments = "-NoLogo -$input_apartment -NoProfile -NonInteractive -ExecutionPolicy $input_executionPolicy -File `"$filePath`""
+    $splat = @{
+        'FileName' = $powershellPath
+        'Arguments' = $arguments
+        'WorkingDirectory' = $workingDirectory
```
Comment (TingluoHuang) [inline Tasks/powerShell/powershell.ps1]: input_workingDirectory

**Comment by ericsciple** [comment]
note to self, finished testing win impl

**Comment by kzu** [comment]
Heya, 
is it intentional that the inline script version does *not* provide Arguments?

This significantly reduces its usefulness when used in task groups that can automatically discover configurable inputs from arguments used in a script.

image

vs File Path:

image

**Pull request merged**